diff --git a/drupal/sites/all/modules/contrib/boxes/boxes.module b/drupal/sites/all/modules/contrib/boxes/boxes.module
index f2986cb..aa4c42f 100644
--- a/drupal/sites/all/modules/contrib/boxes/boxes.module
+++ b/drupal/sites/all/modules/contrib/boxes/boxes.module
@@ -281,20 +281,20 @@ function boxes_footer() {
       if (!empty($errors)) { /* validation error */
         $msg_and_form = theme('status_messages') . drupal_render($form);
         $output[] = array(
-          'command' => 'showBoxForm', 
-          'method' => 'html', 
-          'selector' => '#block-boxes-' . $delta, 
-          'data' => $msg_and_form, 
+          'command' => 'showBoxForm',
+          'method' => 'html',
+          'selector' => '#block-boxes-' . $delta,
+          'data' => $msg_and_form,
           'settings' => NULL,
         );
       }
       //  if we are comming from a init form and need to continue then lets render the form again
       else if (empty($form_state['submitted']) || isset($_GET['init_form_continue'])) {
         $output[] = array(
-          'command' => 'showBoxForm', 
-          'method' => 'html', 
-          'selector' => '#boxes-box-' . $delta, 
-          'data' => drupal_render($form), 
+          'command' => 'showBoxForm',
+          'method' => 'html',
+          'selector' => '#boxes-box-' . $delta,
+          'data' => drupal_render($form),
           'settings' => NULL,
         );
       }
@@ -480,15 +480,15 @@ function boxes_box_form($form, &$form_state) {
  */
 function boxes_form_alter(&$form, &$form_state, $form_id) {
   $path = implode('/', array_slice(arg(), 0, 5));
-  if (($form_id != 'block_admin_configure' || $path != 'admin/structure/block/manage/boxes') && 
+  if (($form_id != 'block_admin_configure' || $path != 'admin/structure/block/manage/boxes') &&
       ($form_id != 'boxes_add_form' || arg(3) != 'box-add')) {
     return;
   }
-  
+
   // Use the Box form submit handler and properly redirect to the correct admin page
   array_unshift($form['#submit'], 'boxes_box_form_submit');
   $form['#action'] = '?destination=admin/structure/block';
-  
+
   // Cancel behavior is different here
   unset($form['cancel']['#attributes']);
   $form['cancel']['#submit'] = array('boxes_block_cancel_submit');
@@ -510,7 +510,7 @@ function boxes_box_form_submit($form, $form_state) {
   // Let's set the additional_classes as an option
   $box->options['additional_classes'] = isset($form_state['values']['additional_classes']) ?
     $form_state['values']['additional_classes'] : array();
-    
+
   if (module_exists('spaces') && $space = spaces_get_space()) {
     $space->controllers->boxes->set($box->delta, $box);
   }
@@ -539,20 +539,20 @@ function boxes_form_block_admin_configure_alter(&$form, &$form_state) {
         '#submit' => array('boxes_block_delete_submit'),
       );
     }
-    
+
     // Cancel behavior is different on admin/structure page
     unset($form['settings']['cancel']);
     $form['actions']['cancel'] = array(
       '#type' => 'submit',
       '#value' => t('Cancel'),
       '#submit' => array('boxes_block_cancel_submit'),
-    );    
+    );
   }
 }
 
 /**
  * Submit handler for box cancel from the block/manage/boxes/[MACHINE-NAME]/configure
- */ 
+ */
 function boxes_block_cancel_submit($form, &$form_state) {
   if (module_exists('overlay')) {
     overlay_close_dialog();
@@ -813,7 +813,15 @@ function boxes_preprocess_block(&$vars) {
   if ($vars['block']->module === 'boxes') {
   }
   if ($vars['block']->module === 'boxes' && isset($vars['block']->boxes_plugin)) {
-    $vars['classes_array'][] = 'block-boxes-' . $vars['block']->boxes_plugin;
+    if (empty($vars['block']->additional_classes)) {
+      return;
+    }
+
+    $additional_classes = explode(' ', $vars['block']->additional_classes);
+
+    foreach ($additional_classes as $class) {
+      $vars['classes_array'][] = $class;
+    }
   }
   if ($vars['block']->module === 'boxes' && isset($vars['block']->additional_classes)) {
     $vars['classes_array'] = array_merge($vars['classes_array'], explode(' ', $vars['block']->additional_classes));
diff --git a/drupal/sites/all/modules/contrib/boxes/plugins/boxes_simple.inc b/drupal/sites/all/modules/contrib/boxes/plugins/boxes_simple.inc
index 09b95c0..b09d79a 100644
--- a/drupal/sites/all/modules/contrib/boxes/plugins/boxes_simple.inc
+++ b/drupal/sites/all/modules/contrib/boxes/plugins/boxes_simple.inc
@@ -4,6 +4,8 @@
  * Simple custom text box.
  */
 class boxes_simple extends boxes_box {
+  const BOXES_EDIT_IN_PLACE = 1;
+
   /**
    * Implementation of boxes_box::options_defaults().
    */
@@ -13,6 +15,7 @@ class boxes_simple extends boxes_box {
         'value' => '',
         'format' => filter_default_format(),
       ),
+      'additional_classes' => '',
     );
   }
 
@@ -20,9 +23,22 @@ class boxes_simple extends boxes_box {
    * Implementation of boxes_box::options_form().
    */
   public function options_form(&$form_state) {
-    $format = filter_format_load($this->options['body']['format']);
+    $option_format = isset($this->options['body']['format']) ? $this->options['body']['format'] : 'plain_text';
+    $format = filter_format_load($option_format);
 
-    if (filter_access($format)) {
+    // TODO: this currently breaks if you switch mode and try to edit existing
+    // simple boxes in the other mode. DO NOT USE IN PRODUCTION /!\
+    if (variable_get('boxes_edit_location', BOXES_EDIT_IN_PLACE) == BOXES_EDIT_IN_PLACE) {
+      $form['body'] = array(
+        '#type' => 'textarea',
+        '#title' => t('Box body'),
+        '#default_value' => $this->options['body'],
+        '#rows' => 6,
+        '#format' => '',
+        '#description' => t('The content of the block as shown to the user.'),
+      );
+    }
+    elseif (filter_access($format)) {
       $form = array();
       $form['body'] = array(
         '#type' => 'text_format',
@@ -33,8 +49,21 @@ class boxes_simple extends boxes_box {
         '#format' => $this->options['body']['format'] ? $this->options['body']['format'] : NULL,
         '#description' => t('The content of the block as shown to the user.'),
       );
-      return $form;
+      $form['view_advanced_settings'] = array(
+        '#type' => 'fieldset',
+        '#title' => t('Advanced settings'),
+        '#collapsed' => TRUE,
+        '#collapsible' => TRUE
+      );
+      $form['view_advanced_settings']['additional_classes'] = array(
+        '#title' => t('Additional CSS classes'),
+        '#description' => t('Optional CSS classes that will be added to the top-level div container for this box. Separate them with spaces.'),
+        '#type' => 'textfield',
+        '#default_value' => $this->options['additional_classes'],
+      );
     }
+
+    return $form;
   }
 
   /**
@@ -42,15 +71,29 @@ class boxes_simple extends boxes_box {
    */
   public function render() {
     $content = '';
-    if (!empty($this->options['body']['value']) && isset($this->options['body']['format'])) {
-      $content = check_markup($this->options['body']['value'], $this->options['body']['format'], $langcode = '' /* TODO Set this variable. */, FALSE);
+
+    if (!empty($this->options['body']['value'])) {
+      if (isset($this->options['body']['format']) && !$this->options['body']['format']) {
+        $content = check_markup($this->options['body']['value'], $this->options['body']['format'], $langcode = '' /* TODO Set this variable. */, FALSE);
+      }
+      else {
+        $content = $this->options['body'];
+      }
     }
+
     $title = isset($this->title) ? $this->title : NULL;
-    return array(
+    $box = array(
       'delta' => $this->delta, // Crucial.
       'title' => $title,
       'subject' => $title,
       'content' => $content,
     );
+
+    // Additional Classes
+    if (!empty($this->options['additional_classes'])) {
+      $box['additional_classes'] = $this->options['additional_classes'];
+    }
+
+    return $box;
   }
 }
