diff --git a/context_ui/export_ui/context_export_ui.class.php b/context_ui/export_ui/context_export_ui.class.php
index c5a4b1a..8c899f5 100644
--- a/context_ui/export_ui/context_export_ui.class.php
+++ b/context_ui/export_ui/context_export_ui.class.php
@@ -102,7 +102,7 @@ function context_ui_form(&$form, &$form_state) {
   // Core context definition
   $form['info']['#type'] = 'fieldset';
   $form['info']['#tree'] = FALSE;
-  $form['info']['name']['#element_validate'] = array('context_ui_edit_name_validate');
+  //$form['info']['name']['#element_validate'] = array('context_ui_edit_name_validate');
 
   $form['info']['tag'] = array(
     '#title' => t('Tag'),
