diff --git a/drupal/sites/all/modules/contrib/menu_attributes/menu_attributes.module b/drupal/sites/all/modules/contrib/menu_attributes/menu_attributes.module
index b4b994a..b8195cd 100644
--- a/drupal/sites/all/modules/contrib/menu_attributes/menu_attributes.module
+++ b/drupal/sites/all/modules/contrib/menu_attributes/menu_attributes.module
@@ -42,7 +42,7 @@ function menu_attributes_menu_link_alter(&$item, $menu) {
  */
 function menu_attributes_menu_attribute_info() {
   $info['title'] = array(
-    'label' => t('Title'),
+    'label' => t('Description'),
     'description' => t('The description displayed when hovering over the link.'),
     'form' => array(
       '#type' => 'textarea',
@@ -51,10 +51,11 @@ function menu_attributes_menu_attribute_info() {
   );
   $info['id'] = array(
     'label' => t('ID'),
-    'description' => t('Specifies a unique ID for the link.'),
+    'description' => t('Specifies a unique ID for the link (example: \'series\')'),
   );
   $info['name'] = array(
-    'label' => t('Name'),
+    'label' => t('More Label'),
+    'description' => t('Text for the more link on MegaMenus'),
   );
   $info['rel'] = array(
     'label' => t('Relationship'),
