diff --git a/drupal/sites/all/modules/contrib/entityreference_autocomplete/includes/autocomplete_callback.inc b/drupal/sites/all/modules/contrib/entityreference_autocomplete/includes/autocomplete_callback.inc
index 595276c..8ea5e19 100644
--- a/drupal/sites/all/modules/contrib/entityreference_autocomplete/includes/autocomplete_callback.inc
+++ b/drupal/sites/all/modules/contrib/entityreference_autocomplete/includes/autocomplete_callback.inc
@@ -66,6 +66,12 @@ function entityreference_autocomplete_autocomplete_callback($entity_type, $bundl
       $query->range(0, $query_settings['limit']);
     }

+    // Sort the results
+    if (!empty($query_settings['property_orderby'])) {
+      $property_orderby = $query_settings['property_orderby'];
+      $query->propertyOrderBy($property_orderby[0], $property_orderby[1]);
+    }
+
     // Execute query and log any unexpected errors.
     try {
       $result = $query->execute();
