diff --git contrib/fb_registration.module contrib/fb_registration.module
index 8c08bc3..7a9cf7e 100644
--- contrib/fb_registration.module
+++ contrib/fb_registration.module
@@ -46,7 +46,7 @@ function fb_registration_form_alter(&$form, &$form_state, $form_id) {
     return;
   }
 
-  if ($form_id == 'user_register') {
+  if ($form_id == 'user_register_form') {
     // Replace regular form with fb_registration.
     $form['#fb_registration'] = TRUE;
 
diff --git fb_user.admin.inc fb_user.admin.inc
index 3c05ad5..824dee2 100644
--- fb_user.admin.inc
+++ fb_user.admin.inc
@@ -61,7 +61,7 @@ function fb_user_admin_settings() {
     '#type' => 'textfield',
     '#title' => t('Registration form button text'),
     '#size' => 60,
-    '#default_value' => _fb_user_button_text('user_register'),
+    '#default_value' => _fb_user_button_text('user_register_form'),
     '#description' => t('Leave blank for facebook\'s default, currently "Login".'),
   );
   $form['fb_user_alter']['login'] = array(
diff --git fb_user.module fb_user.module
index 9857c1f..bc0002a 100644
--- fb_user.module
+++ fb_user.module
@@ -474,7 +474,7 @@ function _fb_user_button_text($form_id) {
 
   if (!isset($button_text)) {
     $button_text = array(
-      'user_register'     => variable_get(FB_USER_VAR_TEXT_REGISTER, NULL),
+      'user_register_form'     => variable_get(FB_USER_VAR_TEXT_REGISTER, NULL),
       'user_login'        => variable_get(FB_USER_VAR_TEXT_LOGIN, NULL),
       'user_login_block'  => variable_get(FB_USER_VAR_TEXT_LOGIN_BLOCK, NULL),
     );
@@ -502,7 +502,7 @@ function fb_user_form_alter(&$form, &$form_state, $form_id) {
   if (isset($GLOBALS['_fb'])) {
     $fb = $GLOBALS['_fb'];
     if (!$GLOBALS['user']->uid && // No alters to user add form.
-        (($form_id == 'user_register' && variable_get(FB_USER_VAR_ALTER_REGISTER, TRUE)) ||
+        (($form_id == 'user_register_form' && variable_get(FB_USER_VAR_ALTER_REGISTER, TRUE)) ||
          ($form_id == 'user_login' && variable_get(FB_USER_VAR_ALTER_LOGIN, TRUE)) ||
          ($form_id == 'user_login_block' && variable_get(FB_USER_VAR_ALTER_LOGIN_BLOCK, TRUE)))) {
 
@@ -524,7 +524,7 @@ function fb_user_form_alter(&$form, &$form_state, $form_id) {
           '#weight' => -1,
         );
 
-        if ($form_id == 'user_register') {
+        if ($form_id == 'user_register_form') {
           // Provide defaults for name and email.
           if ($data = _fb_user_facebook_data($fb)) {
             $form['fb_user']['#fb_user'] = $data;
