diff --git a/googlenews.module b/googlenews.module
index ee37a60..5957c0f 100644
--- a/googlenews.module
+++ b/googlenews.module
@@ -95,13 +95,13 @@ function googlenews_tag_list() {
  * Get a list of all nodes to be output in the Google News sitemap.
  *
  * return array
- *   An array of node nid's, with the 'created' and 'title' values.
+ *   An array of node nid's.
  */
 function googlenews_list_nodes() {
   $time = REQUEST_TIME - intval(variable_get('googlenews_content_hours', 48)) * 3600;
 
   $query = db_select('node', 'n');
-  $query->fields('n', array('nid', 'created', 'title'));
+  $query->fields('n', array('nid'));
   $query->condition('n.type', variable_get('googlenews_node_types', array_keys(node_type_get_names())));
   $query->condition('n.status', NODE_PUBLISHED);
   $query->condition('n.created', $time, '>=');
diff --git a/googlenews.sitemap.inc b/googlenews.sitemap.inc
index a524b05..d42cd45 100644
--- a/googlenews.sitemap.inc
+++ b/googlenews.sitemap.inc
@@ -21,9 +21,6 @@ function googlenews_getgooglenews() {
 
   // If nothing loaded from the cache, build it now.
   if (empty($content)) {
-    // Get a list of all nodes to be output.
-    $nodes = googlenews_list_nodes();
-
     $content = '<?xml version="1.0" encoding="UTF-8"?>';
     $content .= '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"';
     $content .= '  xmlns:news="http://www.google.com/schemas/sitemap-news/0.9">';
@@ -43,18 +40,27 @@ function googlenews_getgooglenews() {
       }
     }
 
-    // Loop through each node.
-    foreach ($nodes as $node) {
+    // Loop through each record.
+    foreach (googlenews_list_nodes() as $record) {
+      // Load the node.
+      $node = node_load($record->nid);
+
+      // Get the node's language, use the global language string if the node
+      // does not have one assigned.
+      $langcode = entity_language('node', $node);
+      if ($langcode == LANGUAGE_NONE) {
+        $langcode = $GLOBALS['language']->language;
+      }
+
       $content .= '<url>';
       $content .= '<loc>' . url('node/' . $node->nid, array('absolute' => TRUE)) . '</loc>';
       $content .= '<news:news>';
       $content .= '<news:publication>';
       $content .= '<news:name>' . $publication_name . '</news:name>';
-      $content .= '<news:language>' . check_plain(entity_language('node', $node)) . '</news:language>';
+      $content .= '<news:language>' . check_plain($langcode) . '</news:language>';
 
       // Insert any added tags if they have values; processing tokens first.
       if (!empty($tags)) {
-        $node = node_load($node->nid);
         foreach ($tags as $tag => $value) {
           // Some items can be output as a comma-separated list.
           if (is_array($value)) {
