diff --git a/drupal/sites/all/modules/contrib/entityreference/entityreference.module b/drupal/sites/all/modules/contrib/entityreference/entityreference.module
index bdcb562..b4381f1 100644
--- a/drupal/sites/all/modules/contrib/entityreference/entityreference.module
+++ b/drupal/sites/all/modules/contrib/entityreference/entityreference.module
@@ -1049,7 +1049,7 @@ function entityreference_autocomplete_callback_get_matches($type, $field, $insta

   if (isset($tag_last)) {
     // Get an array of matching entities.
-    $entity_labels = $handler->getReferencableEntities($tag_last, $instance['widget']['settings']['match_operator'], 10);
+    $entity_labels = $handler->getReferencableEntities($tag_last, $instance['widget']['settings']['match_operator']);

     // Loop through the products and convert them into autocomplete output.
     foreach ($entity_labels as $values) {
