diff --git a/drupal/sites/all/modules/contrib/nodequeue/nodequeue.module b/drupal/sites/all/modules/contrib/nodequeue/nodequeue.module
index 5f84cd5..101823d 100644
--- a/drupal/sites/all/modules/contrib/nodequeue/nodequeue.module
+++ b/drupal/sites/all/modules/contrib/nodequeue/nodequeue.module
@@ -1796,6 +1796,9 @@ function nodequeue_api_autocomplete($queue, $subqueue, $string) {
     $query->where('LOWER(n.title) LIKE LOWER(:string)', array(':string' => '%' . db_like($string) . '%'));
   }
 
+  //Autocomplete Sorting Order
+  $query ->orderBy('created', 'DESC');
+
   // Call to the API.
   $function = $queue->owner . "_nodequeue_autocomplete";
   if (function_exists($function)) {
