diff --git a/drupal/.htaccess b/drupal/.htaccess
index 246edc2..ca30ad8 100644
--- a/drupal/.htaccess
+++ b/drupal/.htaccess
@@ -127,5 +127,10 @@ DirectoryIndex index.php index.html index.htm
       # Force proxies to cache gzipped & non-gzipped css/js files separately.
       Header append Vary Accept-Encoding
     </FilesMatch>
+
+    <FilesMatch "\.(eot|otf|woff|ttf)$">
+      SetEnvIf Origin "^http(s)?://(.+\.)?(takepart\.com)$" origin_is=$0
+      Header set Access-Controll-Allow-Origin %{origin_is}e env=origin_is
+    </FilesMatch>
   </IfModule>
 </IfModule>
