diff --git a/media.module b/media.module
index b6caf7e..06a24dc 100644
--- media.module
+++ media.module
@@ -1110,3 +1110,34 @@ function media_ctools_plugin_api($owner, $api) {
     return array('version' => $api_versions[$owner][$api]);
   }
 }
+
+/**
+ * Implements hook_field_views_data().
+ *
+ * In addition to the default field information we add the relationship for
+ * views to connect back to the file_manage table.
+ */
+function media_field_views_data($field) {
+  // No module_load_include(): this hook is invoked from
+  // views/modules/field.views.inc, which is where that function is defined.
+  $data = field_views_field_default_views_data($field);
+ 
+  $storage = $field['storage']['details']['sql'];
+  foreach ($storage as $age => $table_data) {
+    $table = key($table_data);
+    $columns = current($table_data);
+    $id_column = $columns['fid'];
+    if (isset($data[$table])) {
+      $data[$table][$id_column]['relationship'] = array(
+        'handler' => 'references_handler_relationship',
+        'base' => 'file_managed',
+        'field' => 'fid',
+        'base field' => 'fid',
+        'label' => $field['field_name'],
+        'field_name' => $field['field_name'],
+      );
+    }
+  }
+  return $data;
+}
+ 
